### PR TITLE
Reset operations for each resource class and remove api prefix to mak…

### PR DESCRIPTION
**Pull request by samvdb**
| Q             | A
| ------------- | ---
| Bug fix?      | yes
| New feature?  | no
| BC breaks?    | no
| Deprecations? | no
| Tests pass?   | yes
| Fixed tickets | 
| License       | MIT
| Doc PR        | 

This PR fixes a problem where the generated swagger output was wrong because some resources got tagged incorrectly.

Example:

$operation['item'] and $operation['collection'] are not empty and the next resource has no collection operations defined in it's annotation. Only $operation['item'] will be overriden and thus the swagger documentation would contain output from the previous resource in its collection array.

When prefixing the annotation platform resources like this:

```xml
<import resource="." type="api_platform" prefix="/api"/>
```

Generated paths in swagger will look like /api/foo and the basepath will be /api for the hydra doc route.
Swagger javascript will construct the url to call like this: /api/api/foo which is invalid.

### CODE REVIEW

**Comment by samvdb** [comment]
Not sure what changed in the swagger fixtures but my swagger file validates just fine with this code.

**Comment by dunglas** [comment]
Thanks for testing and fixing this!

It's the Swagger file from the test app. You can generate it to debug by running: `tests/Fixtures/app/console api:swagger:export > swagger.json`

Ping @Simperfit

File: src/Swagger/ApiDocumentationBuilder.php [hunk 1]
```diff
@@ -79,13 +79,14 @@ public function getApiDocumentation() : array
     {
         $classes = [];
         $operation = [];
-        $operation['item'] = [];
-        $operation['collection'] = [];
 
         $itemOperationsDocs = [];
         $definitions = [];
+        $basePath = $this->urlGenerator->generate('api_hydra_entrypoint');
```
Comment (dunglas) [inline src/Swagger/ApiDocumentationBuilder.php]: Can't you pass the `ApiPlatform\Core\Api\UrlGeneratorInterface::REL_PATH);` as second parameter and remove the addition at line 167 instead?

Comment (dunglas) [inline src/Swagger/ApiDocumentationBuilder.php]: Nevermind, it will not work.

File: src/Swagger/ApiDocumentationBuilder.php [hunk 2]
```diff
@@ -79,13 +79,14 @@ public function getApiDocumentation() : array
...
 
         foreach ($this->resourceNameCollectionFactory->create() as $resourceClass) {
+            $operation['item'] = [];
```
Comment (Simperfit) [inline src/Swagger/ApiDocumentationBuilder.php]: why this change ?

Comment (samvdb) [inline src/Swagger/ApiDocumentationBuilder.php]: To reset the operation collection.
See my example posted above.

**Comment by samvdb** [comment]
@Simperfit how do i fix the swagger json in travis?

**Comment by dunglas** [comment]
@samvdb The Swagger file generated after your patch isn't valid anymore.

1. Install the Swagger validator: `npm install -g swagger-cli`
2. Then  generate the Swagger file of the test app by running:`tests/Fixtures/app/console api:swagger:export > swagger.json`
3. Finally run `swagger validate swagger.json` to validate the generate file and debug

**Comment by Simperfit** [comment]
I'm currently looking at the swagger file generated after the patch you made

**Comment by samvdb** [comment]
Yeah i see it now, doesn't seem to be a problem i introduced though :)

**Comment by Simperfit** [comment]
@samvdb It is : 
You change the generation of the path  : 
 `\/users/\{id}` became `"users\/{id}":` which is wrong according to the [specification](http://swagger.io/specification/#pathItemObject).

**Comment by dunglas** [comment]
I propose an easier fix: remove the `basePath` parameter totally. It is optional according to the Swagger spec, and as we generate URLs starting with a `/`, it will fix the problem.

**Comment by samvdb** [comment]
Still doesnt validates my swagger file.

➜  DunglasApiBundle git:(patch-1-swagger) ✗ tests/Fixtures/app/console api:swagger:export > swagger.json
PHP Fatal error:  A function with return type must return a value in Swagger/ApiDocumentationBuilder.php on line 343

**Comment by dunglas** [comment]
@samvdb can you provide the full stack trace, it's probably another bug.

**Comment by dunglas** [comment]
@samvdb, indeed, line 343 must return an empty array, not void like now. Can you change that too?

**Comment by samvdb** [comment]
Already changed it, still waiting for tests to pass.

File: src/Swagger/ApiDocumentationBuilder.php [hunk 3]
```diff
@@ -128,7 +125,7 @@ public function getApiDocumentation() : array
                 }
 
                 $range = $this->getRange($propertyMetadata);
-                if (null === $range) {
+                if (count($range) === 0) {
```
Comment (dunglas) [inline src/Swagger/ApiDocumentationBuilder.php]: `empty($range)` is more performant.

**Comment by dunglas** [comment]
Can you rebase?

**Comment by samvdb** [comment]
@dunglas all good now

**Comment by Simperfit** [comment]
LGTM 👍

**Comment by dunglas** [comment]
Thank you very much @samvdb!
